Jenkinsfile

```diff
@@ -15,22 +15,53 @@ pipeline {
         }
     }
     stages {
-        stage('Install') { 
+        stage('Install system-wide dependencies') { 
+            steps {
+                sh 'apt-get update -y'
+                sh 'apt install curl git -y'   
+            }
+        }
+        stage('Install Python dependencies') { 
             steps {
                 sh('python --version')
                 sh('pip install --user pipenv')
                 sh '$HOME/.local/bin/pipenv lock --dev --requirements > requirements.txt' 
                 sh 'pip install --user -r requirements.txt'
-                sh 'apt-get update -y'
-                sh 'apt install curl -y'
             }
         }
-         stage('Test') { 
+        stage('Test') { 
             steps {
                 sh 'python -m pytest --cov=torrent_tracker_scraper/ --cov-report xml' 
             }
         }
-        stage('Upload Coverage badge') { 
+        stage('Generate changelog') { 
+            environment {
+                VERSION = """${sh(
+                returnStdout: true,
+                script: 'python -m bump'
+                )}"""
+            }
+            when {
+                branch "develop"
+            }
+            steps {
+                sh 'echo ${VERSION}'
+                sh 'git config user.name jenkins'
+                sh 'git config user.email kenokech94@gmail.com'
+                sh 'git stash save setup.py'
```

Comment:
Could you tell me why you stash changes here?

Reply:
The real question is why is `develop` checked out twice? Nice catch.

https://github.com/project-mk-ultra/torrent-tracker-scraper/blob/dfa0794d7c3b47844acf6281f32db0a493a41d9a/Jenkinsfile#L52

Reply:
I remember now. This code is not running on Jenkins. It's running inside the Docker container which doesn't have a repo.